src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -26,13 +27,107 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
         public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, Uri endpoint, Logger logger, CancellationToken cancellationToken)
         {
+            _isMicrosoftSchema = false;
+
             FeatureFlag featureFlag = ParseFeatureFlag(setting.Key, setting.Value);
 
             var keyValues = new List<KeyValuePair<string, string>>();
 
+            if (_isMicrosoftSchema)
+            {
+                keyValues = ProcessMicrosoftSchemaFeatureFlag(featureFlag, setting, endpoint);
+            }
+            else
+            {
+                keyValues = ProcessDotnetSchemaFeatureFlag(featureFlag, setting, endpoint);
+            }
+
+            return Task.FromResult<IEnumerable<KeyValuePair<string, string>>>(keyValues);
+        }
+
+        public bool CanProcess(ConfigurationSetting setting)
+        {
+            string contentType = setting?.ContentType?.Split(';')[0].Trim();
+
+            return string.Equals(contentType, FeatureManagementConstants.ContentType) ||
+                                 setting.Key.StartsWith(FeatureManagementConstants.FeatureFlagMarker);
+        }
+
+        public bool NeedsRefresh()
+        {
+            return false;
+        }
+
+        public void OnChangeDetected(ConfigurationSetting setting = null)
+        {
+            return;
+        }
+
+        public void OnConfigUpdated()
+        {
+            _featureFlagIndex = 0;
+
+            return;
+        }
+
+        private List<KeyValuePair<string, string>> ProcessDotnetSchemaFeatureFlag(FeatureFlag featureFlag, ConfigurationSetting setting, Uri endpoint)
+        {
+            var keyValues = new List<KeyValuePair<string, string>>();
+
```

Comment:
`ProcessMicrosoftSchemaFeatureFlag` does 

```

            if (string.IsNullOrEmpty(featureFlag.Id))
            {
                return keyValues;
            }
```

I'd suggest we unify the approach.